astropy/table/table.py

```diff
@@ -3274,10 +3275,21 @@ def insert_row(self, index, vals=None, mask=None):
         else:
             raise TypeError("Vals must be an iterable or mapping or None")
 
+        @contextmanager
+        def reraise_true_exceptions_as_ValueError(name: str):
+            with warnings.catch_warnings():
```

Comment:
You're right that warnings are not popping up at all now, when they should still cause tests to fail, so my approach isn't even working (yet). However it is not clear to me that ignoring the warning in tests is worth it, and if I need to change 157 tests to workaround it, it feels like a good indication that the new warning will affect many users and might cause a lot of pain downstream. I'll give myself about 30min to study exactly how large of an impact this would do in the test suite and report here.